src/main.rs

```diff
@@ -114,11 +114,40 @@ async fn session_info(
     host_token: String,
     config: &State<Config>,
     db: SessionDBConn,
+    token: TokenCookie,
 ) -> Result<RenderedCredentials, Error> {
-    let credentials = get_credentials_for_host(host_token, config, db)
-        .await
-        .unwrap_or_else(|_| Vec::new());
-    render_credentials(credentials, CredentialRenderType::Json)
+    if check_token(token, config).await? {
+        let credentials = get_credentials_for_host(host_token, config, db)
+            .await
+            .unwrap_or_else(|_| Vec::new());
+        return render_credentials(credentials, CredentialRenderType::Json);
+    }
+
+    Err(Error::Forbidden(
+        "Insufficient rights, try logging in to another account",
+    ))
+}
+
+#[get("/session_info/<host_token>", rank = 2)]
+async fn session_info_anon(host_token: String, config: &State<Config>) -> Result<(), Error> {
+    let login_url = format!("{}/oauth/login?redirect=/logged_in", config.external_url());
+    Err(Error::Unauthorized(login_url))
+}
+
+#[get("/logged_in")]
+async fn logged_in(config: &State<Config>, token: TokenCookie) -> Result<String, Error> {
+    if check_token(token, config).await? {
+        return Ok("You can close this window".to_owned());
```

Comment:
These strings seem to be user visible, so we should make them localizable in my view.

Reply:
I agree, but there is currently not a lot of support for localization. We could use (a combination of):
https://github.com/Plume-org/rocket_i18n
https://github.com/Koka/gettext-rs/
https://github.com/projectfluent/fluent-rs/